res/ps_border.fs.glsl

```diff
@@ -58,10 +58,28 @@ vec4 draw_dotted_edge() {
   return mix(white, circleColor, circleColor.a);
 }
 
-vec4 draw_dashed_edge() {
+// Our current edge calculation is based only on
+// the size of the border-size, but we need to draw
+// the dashes in the center of the segment we're drawing.
+// This calculates how much to nudge and which axis to nudge on.
+vec2 get_dashed_nudge_factor(vec2 dash_size, bool is_corner) {
+  if (is_corner) {
+    return vec2(0.0, 0.0);
+  }
+
+  bool xAxisFudge = vBorders.z > vBorders.w;
+  if (xAxisFudge) {
+    return vec2(dash_size.x / 2.0, 0);
+  } else {
+    return vec2(0.0, dash_size.y / 2.0);
+  }
+}
+
+vec4 draw_dashed_edge(bool is_corner) {
   // Everything here should be in device pixels.
   // We want the dot to be roughly the size of the whole border spacing
-  float dash_interval = min(vBorders.w, vBorders.z) * 3;
+  // 5.5 here isn't a magic number, it's just what mostly looks like FF/Chrome
+  float dash_interval = min(vBorders.w, vBorders.z) * 5.5;
```

Comment:
Would be nice to eventually look in the Gecko/WebKit source and see precisely what they do